services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers.py

```diff
@@ -428,6 +437,71 @@ async def pull_input_ports(
     return transferred_bytes
 
 
+@containers_router.post(
+    "/containers/directory-watcher:disable",
+    summary="Disable directory-watcher event propagation",
+    response_model=None,
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+async def disable_directory_watcher(
+    app: FastAPI = Depends(get_application),
+) -> Response:
+    directory_watcher.disable_directory_watcher(app)
+    # SEE https://github.com/tiangolo/fastapi/issues/2253
+    return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+
+@containers_router.post(
+    "/containers/directory-watcher:enable",
+    summary="Enable directory-watcher event propagation",
+    response_model=None,
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+async def enable_directory_watcher(app: FastAPI = Depends(get_application)) -> Response:
+    directory_watcher.enable_directory_watcher(app)
+    # SEE https://github.com/tiangolo/fastapi/issues/2253
+    return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+
+@containers_router.post(
+    "/containers/ports/outputs:create-dirs",
+    summary="Creates the output directories declared by the labels",
```

Comment:
just wondering, why do I need to explicitely call this endpoint? could the sidecar not do it before running the service?

Reply:
Took me a while to figure it out again. Will also update the docstring.

You do not have access to the labels of the image in the sidecar. But you have access to them in the director-v2. That's why this API exists here.